### PR TITLE
Edit order

### PR DESCRIPTION
**Pull request by enricostano**
2016-12-30-180231_2556x1544_scrot

### CODE REVIEW

File: src/containers/Group/Orders/Order/Edit/Cart/Base.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    return (
+      <div>
+        <div>
+          {orderLines && orderLines.map((orderLine) => <Item key={orderLine.id} orderLine={orderLine} />)}
+        </div>
+        <div>
+          Total {grandTotal}
+        </div>
+      </div>
```
Comment (erisalke) [inline src/containers/Group/Orders/Order/Edit/Cart/Base.js:22]:
```javascript
return (
<div>
{
orderLines && orderLines.map( orderLine => 
<Item
	key={orderLine.id}
	orderLine={orderLine}
/>
)}
<hr />
Total {grandTotal}
</div>
);
```

File: src/containers/Group/Orders/Order/Edit/Products/Base.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        <thead>
+          <tr>
+            <th>Nombre</th>
+            <th>Precio</th>
+            <th>Unidad</th>
+            <th>Cantidad</th>
+          </tr>
+        </thead>
+        <tbody>
+          {activeProducts.map((product) => <Item key={product.id} product={product} order={order}/>)}
```
Comment (erisalke) [inline src/containers/Group/Orders/Order/Edit/Products/Base.js:31]:
```javascript
{
	activeProducts.map( product =>
		<Item key={product.id} product={product} order={order}/> )
}
```
Also `order={order}` is probably `undefined` in here. Check line 51.

File: src/containers/Group/Orders/Order/Edit/Products/Base.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        </tbody>
+      </table>
+    );
+  }
+}
+
+const mapStateToProps = (state, props) => {
+  const activeProducts = state.productsReducer.products.entities.filter((product) => {
+    return _.contains(_.pluck(suppliers, 'producer_id'), product.producer_id);
+  });
```
Comment (erisalke) [inline src/containers/Group/Orders/Order/Edit/Products/Base.js:41]: good candidate for a selector

File: src/containers/Group/Orders/Order/Edit/Products/Base.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    activeProducts: _.map(activeProducts, (product) => {
+      const orderLine = orderLinesByProductId[product.id];
+
+      return {
+        ...product,
+        quantity: orderLine && orderLine.quantity || 0,
+        orderline_id: orderLine && orderLine.id,
+      };
+    }),
+  };
```
Comment (erisalke) [inline src/containers/Group/Orders/Order/Edit/Products/Base.js:54]: I suggest to extract the `_.map( /* ... */ )` out of `return { };`
Also, line 46 pulls an object from state and passes only the orderline.id to props. 
+
in 31: <Item key={product.id} product={product} order={order}

File: src/containers/Group/Orders/Order/Edit/Products/Base.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+  return {
+    activeProducts: _.map(activeProducts, (product) => {
+      const orderLine = orderLinesByProductId[product.id];
+
+      return {
+        ...product,
+        quantity: orderLine && orderLine.quantity || 0,
+        orderline_id: orderLine && orderLine.id,
+      };
```
Comment (erisalke) [inline src/containers/Group/Orders/Order/Edit/Products/Base.js:52]: In line 31:
```
activeProducts.map( product =>
<Item key={product.id} product={product} order={order /* ?? */ }/>)
```

File: src/containers/Group/Orders/Order/Edit/Products/Item/index.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+  constructor(props) {
+    super(props);
+
+    this.handleSubmit = this._handleSubmit.bind(this);
+  }
+
+  _handleSubmit(fields) {
+    const { order, editOrderLine, createOrderLine, destroyOrderLine } = this.props;
+    const { productId, orderLineId } = fields;
```
Comment (erisalke) [inline src/containers/Group/Orders/Order/Edit/Products/Item/index.js]: orderLineId = 0

File: src/containers/Group/Orders/Order/Edit/Products/Item/index.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    const quantity = fields[`quantity_${productId}`];
+
+    if (!orderLineId) {
+      return createOrderLine({ order_id: order.id, product_id: productId, quantity });
+    }
+
+    if (quantity === '0') {
```
Comment (erisalke) [inline src/containers/Group/Orders/Order/Edit/Products/Item/index.js:33]:
```javascript
0 === '0' // number === string
```
1.000.000$ typo 💰 or super hacking 🐛

File: src/redux/modules/order_lines/index.js [hunk 8]
```diff
@@ -105,6 +108,23 @@ export default function orderLinesReducer(state = initialState, action = {}) {
         errors: null,
       };
 
+
+    case DELETE_ORDER_LINE_SUCCESS:
+      entities = _.reject(state.orderLines.entities, (orderLine) => {
+        return orderLine.id === action.requestData.id;
+      });
```
Comment (erisalke) [inline src/redux/modules/order_lines/index.js:115]: `_.reject(state.orderLines.entities, orderLine => orderLine.id === action.requestData.id });`

File: src/containers/Group/Orders/Order/OrderLines/Base.js [hunk 9]
```diff
@@ -5,15 +5,15 @@ import List from './List';
 class OrderLinesBase extends Component {
   static propTypes = {
     orderLines: PropTypes.array.isRequired,
-    grandTotal: PropTypes.number.isRequired,
   }
 
   render() {
-    const { orderLines, grandTotal } = this.props;
+    console.log('PINTATE');
```
Comment (erisalke) [inline src/containers/Group/Orders/Order/OrderLines/Base.js:11]: 🔥

File: src/containers/Group/Orders/Order/OrderLines/Base.js [hunk 10]
```diff
@@ -5,15 +5,15 @@ import List from './List';
...
+    const { orderLines } = this.props;
 
     return (
       <div>
```
Comment (erisalke) [inline src/containers/Group/Orders/Order/OrderLines/Base.js:15]: `<div> </div>` not needed

File: src/containers/Group/Orders/Order/OrderLines/List/index.js [hunk 11]
```diff
@@ -30,7 +26,7 @@ class List extends Component {
           </tr>
         </thead>
         <tbody>
-          {orderLines.map((orderLine) => <Item key={orderLine.id} orderLine={orderLine} />)}
+          {orderLines && orderLines.map((orderLine) => <Item key={orderLine.id} orderLine={orderLine} />)}
```
Comment (erisalke) [inline src/containers/Group/Orders/Order/OrderLines/List/index.js:29]: ditto

File: src/containers/Group/Orders/Order/Show/index.js [hunk 12]
```diff
@@ -1,28 +1,50 @@
...
     params: PropTypes.object.isRequired,
     order: PropTypes.object,
-    orderLines: PropTypes.array,
-    grandTotal: PropTypes.number,
+    orderLines: PropTypes.array.isRequired,
   }
 
   render() {
-    const { order, orderLines, grandTotal } = this.props;
+    const { params: { id }, order, orderLines } = this.props;
```
Comment (erisalke) [inline src/containers/Group/Orders/Order/Show/index.js:21]: cool syntax 👍

File: src/containers/Group/Orders/Order/Edit/Products/Base.js (hunk 3 above)
Comment (enricostano) [reply src/containers/Group/Orders/Order/Edit/Products/Base.js:41]: Yep, thanks! 😍

File: src/containers/Group/Orders/Order/Edit/Products/Item/index.js (hunk 7 above)
Comment (enricostano) [reply src/containers/Group/Orders/Order/Edit/Products/Item/index.js:33]: The problem here is that the input field in the form returns strings, not numbers. But I still need to investigate how to manage with number related input forms 😬

File: src/containers/Group/Orders/Order/OrderLines/Base.js (hunk 9 above)
Comment (enricostano) [reply src/containers/Group/Orders/Order/OrderLines/Base.js:11]: This is a WIP commit, but yes... 🔥 ;)

File: src/containers/Group/Orders/Order/Edit/Products/Item/index.js (hunk 7 above)
Comment (andresgutgon) [reply src/containers/Group/Orders/Order/Edit/Products/Item/index.js:33]: You can do this:

``` js
if (+quantity <= 0) return destroyOrderLine(orderLineId);
```

The `+` is coercing `'0'` into number

Comment (enricostano) [reply src/containers/Group/Orders/Order/Edit/Products/Item/index.js:33]: Oh, will that wotk for decimals too? `3.5` for instance

Comment (andresgutgon) [reply src/containers/Group/Orders/Order/Edit/Products/Item/index.js:33]: Try it ;)

**Comment by sauloperez** [comment]
OMG a new contributor coming soon!

**Comment by enricostano** [comment]
We switched to https://github.com/openfoodfoundation/openfoodnetwork See you there!

**Pull request closed without merging**